test/engine/SpatialJoinTest.cpp

```diff
@@ -0,0 +1,1957 @@
+                                           expectedMaxDist1_rows_diff,
+                                           columnNames, false);
+  buildAndTestSmallTestSetDiffSizeChildren("<max-distance-in-meters:5000>",
+                                           true, expectedMaxDist5000_rows_diff,
+                                           columnNames, true);
+  buildAndTestSmallTestSetDiffSizeChildren("<max-distance-in-meters:5000>",
+                                           false, expectedMaxDist5000_rows_diff,
+                                           columnNames, true);
+  buildAndTestSmallTestSetDiffSizeChildren("<max-distance-in-meters:5000>",
+                                           true, expectedMaxDist5000_rows_diff,
+                                           columnNames, false);
+  buildAndTestSmallTestSetDiffSizeChildren("<max-distance-in-meters:5000>",
+                                           false, expectedMaxDist5000_rows_diff,
+                                           columnNames, false);
+  buildAndTestSmallTestSetDiffSizeChildren(
+      "<max-distance-in-meters:500000>", true, expectedMaxDist500000_rows_diff,
+      columnNames, true);
+  buildAndTestSmallTestSetDiffSizeChildren(
+      "<max-distance-in-meters:500000>", false, expectedMaxDist500000_rows_diff,
+      columnNames, true);
+  buildAndTestSmallTestSetDiffSizeChildren(
+      "<max-distance-in-meters:500000>", true, expectedMaxDist500000_rows_diff,
+      columnNames, false);
+  buildAndTestSmallTestSetDiffSizeChildren(
+      "<max-distance-in-meters:500000>", false, expectedMaxDist500000_rows_diff,
+      columnNames, false);
+  buildAndTestSmallTestSetDiffSizeChildren(
+      "<max-distance-in-meters:1000000>", true,
+      expectedMaxDist1000000_rows_diff, columnNames, true);
+  buildAndTestSmallTestSetDiffSizeChildren(
+      "<max-distance-in-meters:1000000>", false,
+      expectedMaxDist1000000_rows_diff, columnNames, true);
+  buildAndTestSmallTestSetDiffSizeChildren(
+      "<max-distance-in-meters:1000000>", true,
+      expectedMaxDist1000000_rows_diff, columnNames, false);
+  buildAndTestSmallTestSetDiffSizeChildren(
+      "<max-distance-in-meters:1000000>", false,
+      expectedMaxDist1000000_rows_diff, columnNames, false);
+  buildAndTestSmallTestSetDiffSizeChildren(
+      "<max-distance-in-meters:10000000>", true,
+      expectedMaxDist10000000_rows_diff, columnNames, true);
+  buildAndTestSmallTestSetDiffSizeChildren(
+      "<max-distance-in-meters:10000000>", false,
+      expectedMaxDist10000000_rows_diff, columnNames, true);
+  buildAndTestSmallTestSetDiffSizeChildren(
+      "<max-distance-in-meters:10000000>", true,
+      expectedMaxDist10000000_rows_diff, columnNames, false);
+  buildAndTestSmallTestSetDiffSizeChildren(
+      "<max-distance-in-meters:10000000>", false,
+      expectedMaxDist10000000_rows_diff, columnNames, false);
+}
+
+}  // end of Namespace computeResultTest
```

Comment:
I have looked at this until here.
I see you put a lot of effort into those tests which I like very much.
The last part (the manual setting up of inputs)  is okay like this, because it is crucial for the correctness, so it might be a bit longer.

I will continue to look at the remainder (but you can also apply the same principles for your remaining tests).

Reply:
Thanks :)
so the comment from above with the refactoring of the setting up of inputs doesn't apply anymore?